data_transfer/utils/__init__.py

```diff
@@ -47,10 +47,16 @@ def format_weartime_from_timestamp(period: int) -> datetime:
     return datetime.fromtimestamp(period)
 
 
-def get_period_by_days(start: datetime, days: int) -> Tuple[str, str]:
+def get_period_by_days(start: int, days: int) -> Tuple[str, str]:
     """return two timestamps based on a startdate and duration"""
-    end_date = start.replace(hour=23, minute=59)
-    from_date = end_date - timedelta(days=days)
+    reference = datetime.today() - timedelta(days=start)
+
+    # ensure end at night, but start in the morning to capture everything _those days_
+    end_date = reference.replace(hour=23, minute=59, second=59, microsecond=999999)
+    from_date = end_date - timedelta(
+        days=days, hours=23, minutes=59, seconds=59, microseconds=999999
+    )
+
     begin = str(int(from_date.timestamp()))
     end = str(int(end_date.timestamp()))
     return (begin, end)
```

Comment:
Unclear why these are casted to strings? I would return timestamps and cast only where needed, e.g., rather than [casting back to its original type](https://github.com/ideafast/middleware-services/blob/352459b483275c166011d71ff435bf17a1aeb2bb/data_transfer/dags/btf.py#L45-L49)

This would mean that some uses of it would need to be modified, e.g., [here](https://github.com/ideafast/middleware-services/blob/352459b483275c166011d71ff435bf17a1aeb2bb/data_transfer/jobs/byteflies.py#L4)